op-node/rollup/engine/payload_process.go

```diff
@@ -30,21 +30,31 @@ func (eq *EngDeriver) onPayloadProcess(ev PayloadProcessEvent) {
 		ev.Envelope.ExecutionPayload, ev.Envelope.ParentBeaconBlockRoot)
 	if err != nil {
 		eq.emitter.Emit(rollup.EngineTemporaryErrorEvent{
-			Err: fmt.Errorf("failed to insert execution payload: %w", err)})
+			Err: fmt.Errorf("failed to insert execution payload: %w", err),
+		})
 		return
 	}
 	switch status.Status {
 	case eth.ExecutionInvalid, eth.ExecutionInvalidBlockHash:
+		// TODO: I'm not sure how derivation can reach this point. By my understand, it would
+		// already fail earlier during the FCU call.
+		if ev.DerivedFrom != (eth.L1BlockRef{}) && eq.cfg.IsHolocene(ev.DerivedFrom.Time) {
```

Comment:
Regarding the TODO comment: nonce values of a tx are not checked upon FCU I believe, but can result in an invalid tx and invalid block. And similarly balance checks can be a problem.

Reply:
I think the payload is built synchronously if `noTxPool == true`, so I assumed that everything that can invalidate the attributes already did with the FCU call, see [here](https://github.com/ethereum-optimism/op-geth/blob/4539f2d3a77f14bdad362c24e3773dc6aad87d5b/miner/payload_building.go#L281).

Reply:
That's not a guarantee of the engine API though - it may be true for op-geth, but might not be for op-reth (or might change in a future version of op-geth). In fact, I suspect that's a bug in op-geth - I'd have expected block building to always be done in parallel and forkChoiceUpdated should return as fast as possible with any unavoidable delay impacting getPayload instead.

Reply:
Just to clarify - we probably don't need to change op-geth in this case because the noTxPool flag is OP specific and I don't think there's any problem with forkChoiceUpdated being delayed given we will immediately call getPayload in our case (it would be sub-optimal for L1).  But we shouldn't depend on that behaviour.